lib/core/src/Data/Time/Utils.hs

```diff
@@ -23,3 +37,16 @@ utcTimeSucc = addUTCTime $ succ 0
 --   for which 't1 < t0'.
 utcTimePred :: UTCTime -> UTCTime
 utcTimePred = addUTCTime $ pred 0
+
+-- | Suspends the current thread until after the specified time has passed.
+--
+-- There is no guarantee that the thread will be rescheduled promptly when the
+-- delay has expired, but the thread will never continue to run earlier than
+-- specified.
+--
+waitUntil :: MonadIO m => UTCTime -> m ()
+waitUntil futureTime = liftIO $ do
+    currentTime <- getCurrentTime
+    threadDelay
```

Comment:
Does this work OK when the difference is negative?
Perhaps you could put a `max 0` there to make sure.

Reply:
> Does this work OK when the difference is negative?
> Perhaps you could put a `max 0` there to make sure.

I think it should work, as informal testing of `threadDelay` shows that it exits immediately when called with a negative value.

However, just to be sure, I've used `max 0`, as you suggest.

Fixed in bcdeeb8d8327d9b91aacbd6ec71e5545547c4989.